crates/proof-of-sql/src/base/database/table.rs

```diff
@@ -38,15 +45,43 @@ impl<'a, S: Scalar> Table<'a, S> {
     ) -> Result<Self, TableError> {
         Self::try_new(IndexMap::from_iter(iter))
     }
+    /// Creates a new [`Table`] from a [`DataAccessor`], [`TableRef`] and [`ColumnRef`]s.
+    ///
+    /// Columns are retrieved from the [`DataAccessor`] using the provided [`ColumnRef`]s.
+    /// # Panics
+    /// Missing columns or column length mismatches can occur if the accessor doesn't
+    /// contain the necessary columns. In practice, this should not happen.
+    pub(crate) fn from_columns(
```

Comment:
I'd prefer for this to be a provided method on `DataAccessor`, since `Table` doesn't really need to know about the concept of `DataAccessor` to be useful.